src/services/codeFixProvider.ts

```diff
@@ -65,11 +81,15 @@ namespace ts.codefix {
         return createCombinedCodeActions(changes, commands.length === 0 ? undefined : commands);
     }
 
-    export function eachDiagnostic({ program, sourceFile, cancellationToken }: CodeFixAllContext, errorCodes: readonly number[], cb: (diag: DiagnosticWithLocation) => void): void {
-        for (const diag of program.getSemanticDiagnostics(sourceFile, cancellationToken).concat(computeSuggestionDiagnostics(sourceFile, program, cancellationToken))) {
+    export function eachDiagnostic(context: CodeFixAllContext, errorCodes: readonly number[], cb: (diag: DiagnosticWithLocation) => void): void {
+        for (const diag of getDiagnostics(context)) {
             if (contains(errorCodes, diag.code)) {
                 cb(diag as DiagnosticWithLocation);
             }
         }
     }
+
+    function getDiagnostics({ program, sourceFile, cancellationToken }: CodeFixContextBase) {
+        return program.getSemanticDiagnostics(sourceFile, cancellationToken).concat(computeSuggestionDiagnostics(sourceFile, program, cancellationToken));

```

Comment:
Currently, we only support document scope for fix-all, but we left open the option of supporting project or solution.  Is this going to give us mysterious false negatives if we ever light that up?

Reply:
I’m not sure I understand the scenario you’re concerned about, but I don’t think that feature could be enabled without doing work in this area in the first place, since all the code fix context types include a source file.

Reply:
I read this as only looking for diagnostics in the current file.  What if there are diagnostics in other documents that could be corrected with fix-all?

Reply:
Then this would have to change, along with lots of other stuff in the code fix infrastructure—but it’s unrelated to this PR. I just moved this existing functionality into a new function for reuse.